source/s3_auto_ranged_get.c

```diff
@@ -117,6 +119,16 @@ static enum aws_s3_auto_ranged_get_request_type s_s3_get_request_type_for_discov
     struct aws_s3_auto_ranged_get *auto_ranged_get = meta_request->impl;
     AWS_ASSERT(auto_ranged_get);
 
+    /*
+     * If the object is empty, download the file using a GetObject with PartNumber request. This will succeed if
+     * the file is empty, or it will download the file.
+     */
+    if (auto_ranged_get->synced_data.object_range_empty != 0) {
```

Comment:
I'm really confused how we end up back in `s_s3_get_request_type_for_discovering_object_size()` if we've already discovered that `object_range_empty`

When `object_range_empty` becomes true, `object_range_empty` becomes true too
And after `object_range_empty` become true, we don't call `s_s3_get_request_type_for_discovering_object_size()` .. right?

Reply:
I have improved the documentation to add more details.